### PR TITLE
docs: reorganized and pruned button stories

### PR DESCRIPTION
**Pull request by codecaaron**
- Organizes themes by set (predefined, brand, and deprecated).
- Adds default docs mode config to storybook, and removes the `--docs` flag on the production build comment.
- Moves some of the configurations to knobs to be viewed in canvas.

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/brand-components@0.4.2-alpha.8e28b0.3+8e28b0f5<br/>@codecademy/enzyme-helpers@1.1.10-alpha.8e28b0.5+8e28b0f5<br/>@codecademy/gamut-templates@2.3.4-alpha.8e28b0.3+8e28b0f5<br/>@codecademy/gamut@2.7.1-alpha.8e28b0.3+8e28b0f5<br/></pre>

File: packages/styleguide/stories/Button.stories.js [hunk 1]
```diff
@@ -7,8 +7,21 @@ import RadialProgress from '../../gamut/src/RadialProgress';
 import { withKnobs, select, text, boolean } from '@storybook/addon-knobs';
 import { deprecatedColors } from '../../gamut-styles/utils/variables';
 
-const themeKeys = [
-  ...Object.keys(presetThemes),
+const themeKeys = [...Object.keys(presetThemes)];
+
+const brandThemeKeys = [
+  'brand-red',
+  'brand-orange',
```
Comment (JoshuaKGoldberg) [inline packages/styleguide/stories/Button.stories.js]: These actually aren't accessible... https://webaim.org/resources/contrastchecker/?fcolor=A3A3A3&bcolor=FFFFFF, or in the Chrome dev tools:

image

Can they be removed?

The brand color is fine, just not as a button background color.

File: packages/styleguide/stories/Button.stories.js [hunk 2]
```diff
@@ -7,8 +7,21 @@ import RadialProgress from '../../gamut/src/RadialProgress';
 import { withKnobs, select, text, boolean } from '@storybook/addon-knobs';
 import { deprecatedColors } from '../../gamut-styles/utils/variables';
 
-const themeKeys = [
-  ...Object.keys(presetThemes),
+const themeKeys = [...Object.keys(presetThemes)];
```
Comment (JoshuaKGoldberg) [inline packages/styleguide/stories/Button.stories.js]: `primary` and `secondary` also aren't accessible as button colors 🙃

File: packages/styleguide/stories/Button.stories.js [hunk 3]
```diff
@@ -22,111 +35,127 @@ const themeKeys = [
   'ccblue',
   'royalblue',
```
Comment (JoshuaKGoldberg) [inline packages/styleguide/stories/Button.stories.js]: I've been using this `blue` as a drop-in replacement for `blue` in designs, lol. Does this mean I should use `brand-blue` instead?

File: packages/styleguide/package.json [hunk 4]
```diff
@@ -7,7 +7,7 @@
   "scripts": {
     "start": "yarn storybook",
     "storybook": "start-storybook -p 6006 -c .storybook --ci --quiet",
-    "build-storybook": "NODE_ENV=production build-storybook --docs --quiet -c .storybook -o ../../dist/storybook",
```
Comment (jakemhiller) [inline packages/styleguide/package.json]: 🎊

File: packages/styleguide/stories/Button.stories.js (hunk 3 above)
Comment (jakemhiller) [reply packages/styleguide/stories/Button.stories.js]: blah, no, just stick with `blue` for now. We'll swap `blue` out with the new color when Aaron and Yassine figure out what equivalent blue they want.

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/brand-components@0.4.4-alpha.754695.4+754695e5<br/>@codecademy/gamut-templates@2.3.5-alpha.754695.4+754695e5<br/>@codecademy/gamut@2.7.2-alpha.754695.4+754695e5<br/>@codecademy/webpack-config@4.0.1-alpha.754695.9+754695e5<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀Styleguide deploy preview ready!</p><p><a href=https://5e69184f3871357eb1163316--codecademy-styleguide.netlify.com>https://5e69184f3871357eb1163316--codecademy-styleguide.netlify.com</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5e69184f3871357eb1163316>Deploy Logs</a></p>

File: packages/styleguide/stories/Button.stories.js (hunk 1 above)
Comment (codecaaron) [reply packages/styleguide/stories/Button.stories.js]: Probably, I don't think we have an inflight ones for this.  I want to do a pass to remove all the ones that we aren't using at all.

**Pull request merged**